OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -87,6 +87,20 @@ public string GetCursor(World world, CPos xy, MouseInput mi)
 			return cursorName ?? (useSelect ? "select" : "default");
 		}
 
+		// Used for classic mouse orders, determines whether or not action at xy is move or select
+		public static bool InputOverridesSelection(World world, int2 xy, MouseInput mi)
+		{
+			var tar = Target.FromActor(world.ScreenMap.ActorsAt(xy).WithHighestSelectionPriority());
+			var self = world.Selection.Actors.WithHighestSelectionPriority();
+
+			var o = OrderForUnit(self, tar, mi);
+
+			if (o == null || o.Trait is IMove)
```

Comment:
This hardcoded check against `IMove` is a bit iffy, but may be okay for now.  In the future I expect we will need to move this to a specific property that individual order-issuing traits can set to true or false.